.github/workflows/update.yml

```
@@ -0,0 +1,60 @@
+---
+name: Update all puppet modules
+
+on:
+  workflow_dispatch: {}
+  push:
+    tags:
+      - '*'
+
+env:
+  GIT_AUTHOR_NAME: pccibot
+  GIT_AUTHOR_EMAIL: 12855858+pccibot@users.noreply.github.com
+  GIT_COMMITTER_NAME: pccibot
+  GIT_COMMITTER_EMAIL: 12855858+pccibot@users.noreply.github.com
+  SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+  GITHUB_BASE_URL: https://api.github.com
+
+jobs:
+  unit:
+    runs-on: ubuntu-latest
+    name: Run msync against all modules
+    if: github.repository_owner == voxpupuli
+    steps:
+      - name: checkout code
+        # the action has a ssh-key option.
+        # when provided, it configures the ssh/git config to authenticate automatically with that key to the source repo
+        # the authentication won't be used for other repos (which we need for msync), so we cannot use the option
+        # that's the reason for the 'Add SSH key' step
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Do we need the entire history to be fetched? I wouldn't think modulesync cares about that

---

https://github.com/actions/checkout mentions there's `fetch-tags` so I suspect this works as well:
```suggestion
        with:
          fetch-tags: true
```

---

I need to test this, I think that only works when the action runs for a pushed tag, not for the on demand runs.